src/main/java/marquez/service/JobService.java

```diff
@@ -196,8 +200,12 @@ public Job createOrUpdate(
         }
         final JobRow jobRow = jobDao.find(namespaceName.getValue(), jobName.getValue()).get();
         final JobContextRow contextRow = jobContextDao.findBy(checksum).get();
-        final List<UUID> inputUuids = findUuids(jobMeta.getInputs());
-        final List<UUID> outputUuids = findUuids(jobMeta.getOutputs());
+        final List<DatasetRow> inputRows = findUuids(jobMeta.getInputs());
+        final List<UUID> inputUuids =
+            inputRows.stream().map(e -> e.getUuid()).collect(Collectors.toList());
```

Comment:
_Minor:_ You can use `.map(DatasetRow::getUuid)` if you wanted to make the code a bit more compact.